metadata/metadata-modeshape/src/main/java/com/thinkbiganalytics/metadata/modeshape/BaseJcrProvider.java

```diff
@@ -291,16 +329,31 @@ public T findFirst(String query) {
             }
             return null;
         } catch (RepositoryException e) {
-            throw new MetadataRepositoryException("Unable to findAll for Type : " + getNodeType(getJcrEntityClass()), e);
+            throw new MetadataRepositoryException("Unable to findFirst for Type : " + getNodeType(getJcrEntityClass()), e);
         }
     }
 
     @Override
     public List<T> findAll() {
-
-        String jcrQuery = "SELECT * FROM [" + getNodeType(getJcrEntityClass()) + "]";
-        return find(jcrQuery);
-
+        return find(startBaseQuery().toString());
+    }
+    
+    @Override
+    public Page<T> findPage(Pageable pageable, String filter) {
+        int count = findCount(filter);
+        
+        if (count > 0) {
+            StringBuilder bldr = startBaseQuery();
+            appendJoins(bldr, pageable, filter);
+            appendFilter(bldr, filter);
+            appendSort(bldr, pageable);
+            appendOffset(bldr, pageable);
+            
+            List<T> list = find(bldr.toString());
+            return new PageImpl<>(list, pageable, count);
+        } else {
+            return new PageImpl<T>(Collections.emptyList());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)